src/test/scala/akka/persistence/jdbc/migration/V4MigrationSpec.scala

```diff
@@ -0,0 +1,101 @@
+package akka.persistence.jdbc.migration
+
+import java.io.{File, FileOutputStream}
+import java.nio.file.Files
+
+import akka.actor.{ActorSystem, Props}
+import akka.persistence.PersistentActor
+import akka.persistence.jdbc.SimpleSpec
+import akka.persistence.jdbc.config.JournalConfig
+import akka.persistence.jdbc.util.{ClasspathResources, DropCreate, SlickDatabase}
+import com.typesafe.config.ConfigFactory
+import org.h2.util.IOUtils
+import org.scalatest.BeforeAndAfter
+import slick.jdbc.JdbcBackend.Database
+import akka.pattern.ask
+import akka.persistence.jdbc.migraition.V4JournalMigration
+import akka.util.Timeout
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class V4MigrationSpec extends SimpleSpec with BeforeAndAfter with ClasspathResources {
+
+  var system: ActorSystem = _
+  var tmpDirectory: File = _
+
+  before {
+    tmpDirectory = Files.createTempDirectory("akka-persistence-jdbc-migration-").toFile
+    IOUtils.copyAndClose(fromClasspathAsStream("v3-h2-data.mv.db"), new FileOutputStream(new File(tmpDirectory, "database.mv.db")))
```

Comment:
It is probably a good idea to include this test for each supported database.

Reply:
Use `Files.newOutputStream()` see https://www.cloudbees.com/blog/fileinputstream-fileoutputstream-considered-harmful

Reply:
Wouldn't it be possible to insert the rows in legacy format using slick? That would make it database independent